pgjdbc/src/main/java/org/postgresql/util/PGInterval.java

```diff
@@ -367,20 +376,10 @@ public int getMicroSeconds() {
    * @param seconds seconds to set
    */
   public void setSeconds(double seconds) {
-    String str = Double.toString(seconds);
+    String str = String.format(Locale.ROOT,"%.6f",seconds);
```

Comment:
Might be something like
```java
wholeSeconds = (int) seconds
microSeconds  = (seconds - wholeSeconds) / 1_000_000.0;
```

Reply:
Well I have tried the _ thing and sadly java 6 doesn't understand it. Once we dump Java 6 we can use that.
If you really require double then just create a private static final CONSTANT. I'll be fine with that.

Reply:
Surely a constant can be introduced.
I mean that this operation can work only with numbers, without conversion to string.

and while discussing this, seems like there is an issue with getSeconds() which currently does
`Double.parseDouble("" + wholeSeconds + '.' + microSeconds )`
if wholeSeconds=1 and microSeconds=1 then result should be 1.000001 not 1.1

Reply:
I can introduce a constant and fix the `getSeconds` method. I'll just use the conversion logic from `getValue`.

In terms of the `setSeconds` logic, should 9e-7 be rounded nearest to 1 microsecond or can we just cast it down to 0?

Reply:
It would appear we need a few more test cases!